### ISSUE TITLE
Model stops as the training starts.

### ISSUE THREAD

**Issue by Nazzish** [ThilinaRajapakse/simpletransformers#638]
Hello, 

I am using the code in Google Colab and facing an error. As soon as I reach on the following step: 

![image](https://user-images.githubusercontent.com/44400362/89680128-c6b77180-d90b-11ea-8c2e-009c5b9a12f6.png)

the system stops itself and doesn't run further without any error. 
Please let me know what can be the possible issue and how to resolve it?

**Comment by soccerdroid** [comment]
I am having the same issue. Using bert-base-multilingual-cased as a MultiLabelClassificationModel
![image](https://user-images.githubusercontent.com/18273329/89719892-851dd980-d992-11ea-9e7b-47c4d182ffe5.png)

**Comment by soccerdroid** [comment]
any thoughts/hints on this??

**Comment by ThilinaRajapakse** [comment]
> I am having the same issue. Using bert-base-multilingual-cased as a MultiLabelClassificationModel
> ![image](https://user-images.githubusercontent.com/18273329/89719892-851dd980-d992-11ea-9e7b-47c4d182ffe5.png)

Does this remain stuck indefinitely? It's possible that the training is paused for evaluation. Can you check the GPU usage and see if it is working or idle?

**Comment by ThilinaRajapakse** [comment]
> Hello,
> 
> I am using the code in Google Colab and facing an error. As soon as I reach on the following step:
> 
> ![image](https://user-images.githubusercontent.com/44400362/89680128-c6b77180-d90b-11ea-8c2e-009c5b9a12f6.png)
> 
> the system stops itself and doesn't run further without any error.
> Please let me know what can be the possible issue and how to resolve it?

Is this only happening for this dataset or does it happen with any data you use?

**Comment by soccerdroid** [comment]
> > I am having the same issue. Using bert-base-multilingual-cased as a MultiLabelClassificationModel
> > ![image](https://user-images.githubusercontent.com/18273329/89719892-851dd980-d992-11ea-9e7b-47c4d182ffe5.png)
> 
> Does this remain stuck indefinitely? It's possible that the training is paused for evaluation. Can you check the GPU usage and see if it is working or idle?

Yes, I left it for amost an hour I think, and no progress was made. I have cuda disabled.

**Comment by ThilinaRajapakse** [comment]
If CUDA is disabled, then the evaluation might even take longer than an hour even on a moderate dataset.

Can you try running it with CUDA enabled? You can use Colab if needed.

**Comment by Nazzish** [comment]
I have used "cuda_available = torch.cuda.is_available()" as mentioned on your tutorial website on simple transformers. The GPU remains idle at this point shown in the screenshot. I want to use binary classification and have also used models other than bert but the problem persists. I haven't tried other datasets though. 

One more thing, for this setting I had uploaded files to Colab. I also tried by mounting drive. When I run the code by mounting drive and uploading files to the drive, I get the EOF error in read_csv function for both the test and train files. When I add args to read_csv function to ignore the error, it doesn't accept the args.

**Comment by soccerdroid** [comment]
> If CUDA is disabled, then the evaluation might even take longer than an hour even on a moderate dataset.
> 
> Can you try running it with CUDA enabled? You can use Colab if needed.

I tried running it without hyperparameters optimization, and an error with the evaluation metric function popped up. I guess that error was causing the training to hang in. I  fixed it and the training went well (still without sweeping, CUDA disabled). I haven't tried it yet to re run the script with sweeping. Will let you know how it went!

**Comment by Nazzish** [comment]
Any idea how to get it started? The error persists on different datasets as well.

**Comment by cadae** [comment]
Same issue here. This issue started to appear in multilabel classification training after I changed the model's args from dictionary args to ClassificationArgs. And it seems like this issue affects Linux only. The training was able to finish on windows 10 but not on the RedHat 8 with the same dataset and configuration. The workaround would be just using dictionary args for model's args (training args has to be dict). The following args are the one I'm used that stopped the training.
```
model_args = ClassificationArgs(
    regression=False,
    no_cache=True,
    fp16=False,
    silent=False,
    reprocess_input_data=True,
    use_early_stopping=True,
    sliding_window=False,
    save_model_every_epoch=False,
    save_eval_checkpoints=False,
    use_multiprocessing=False,
    cache_dir=MODELS_DIR+"temp",
    tensorboard_dir=MODELS_DIR+"temp",
    train_batch_size=8,
    eval_batch_size=8,
    early_stopping_metric="eval_loss",
    early_stopping_delta=0.01,
    early_stopping_metric_minimize=True,
    early_stopping_patience=3,
    early_stopping_consider_epochs=True,
    labels_map={}
)
```

**Comment by ThilinaRajapakse** [comment]
Can I get a minimal script to reproduce this issue? I have no clue why this is happening.

**Comment by yanqiangmiffy** [comment]
yes,I also have this issue,i see a another cpu process is hanged up

**Comment by creatorrr** [comment]
For now, this has helped me:

```python
model_args = ClassificationArgs(
    use_multiprocessing=False,
    use_multiprocessing_for_evaluation=False,
)

os.environ["TOKENIZERS_PARALLELISM"] = "false"
```

**Comment by gilzeevi25** [comment]
> For now, this has helped me:
> 
> ```python
> model_args = ClassificationArgs(
>     use_multiprocessing=False,
>     use_multiprocessing_for_evaluation=False,
> )
> 
> os.environ["TOKENIZERS_PARALLELISM"] = "false"
> ```

oddly enough, this also helped me so cheers mate

**Comment by nemetht** [comment]
> For now, this has helped me:
> 
> ```python
> model_args = ClassificationArgs(
>     use_multiprocessing=False,
>     use_multiprocessing_for_evaluation=False,
> )
> 
> os.environ["TOKENIZERS_PARALLELISM"] = "false"
> ```

Worked also for me - thank you!

**Comment by phileas-condemine** [comment]
> For now, this has helped me:
> 
> ```python
> model_args = ClassificationArgs(
>     use_multiprocessing=False,
>     use_multiprocessing_for_evaluation=False,
> )
> 
> os.environ["TOKENIZERS_PARALLELISM"] = "false"
> ```

Same here, I've been struggling with this issue for more than a year with unpredictible training staying idle in the middle of the training. 
Thank you for the tip @creatorrr !

**Comment by shtosti** [comment]
> For now, this has helped me:
> 
> ```python
> model_args = ClassificationArgs(
>     use_multiprocessing=False,
>     use_multiprocessing_for_evaluation=False,
> )
> 
> os.environ["TOKENIZERS_PARALLELISM"] = "false"
> ```

This was the only fix that helped me with this problem. Thank you so much for sharing it @creatorrr!

**Comment by StrikerRUS** [comment]
@creatorrr

Thanks a lot for the fix!

**Comment by AyresLucas** [comment]
Thanks a million!
